### DISCUSSION TITLE
Bug: Incorrect evaluation of base currency percentage in X-ray

### DISCUSSION THREAD

**Discussion by Biermannek** [ghostfolio/ghostfolio#2495] [General]
Hello,

I've encountered a possible bug in evaluation of rules in X-ray, more specifically issue with calculation of percentage of investments in my base currency. My base currency is CZK and I have investments in CZK, USD and EUR. If I check distribution of currencies in Allocations sections, then it looks ok and CZK is about 23% (see attached picture). But if I look into X-ray, then it says that I have only 1,26% in CZK (see attached picture).

When I tried rough calculations, then it seems almost like if CZK investment was converted to USD first and then amount in USD was compared with whole portfolio. Right now exchange rate is about 23 CZK for 1 USDso if you divide those 23% by 23 and then take into account that whole investment is now smaller, then calculated 1,26% seems like result which could come from this calculation.

Example based on my distribution between currencies.
USD investment: 64 000 CZK (approx 2782 USD)
EUR investment:  13 000 CZK (approx.  565 USD)
CZK investment: 23 000 CZK (approx. 1000 USD)

I would expect that X-ray will determine that I have 23% in CZK, but I think X-ray does calculation like this:
Divide 23 000 CZK by USD exchange rate (now 23), which leads to 1000 CZK. Then whole investment amount is diminished to 78 000 CZK and if I calculate percentage of 1000 from 78 000, then I get 1,28%. 

Can you please check if there isn't some possible issue like this? Thank you.

![image](https://github.com/ghostfolio/ghostfolio/assets/147999285/0a2a132e-360d-4eee-b2ea-1d3ea6c530fb)
![image](https://github.com/ghostfolio/ghostfolio/assets/147999285/00c18d90-2f12-4b6c-a844-7ee1a34d9e34)

**Comment by dtslvr** [comment]
#2511
